hyperspy/drawing/marker.py

```diff
@@ -77,6 +89,35 @@ def marker_properties(self, kwargs):
             except:
                 pass
 
+    def _to_dictionary(self):
+        marker_dict = {}
+        marker_dict['marker_properties'] = self.marker_properties
+        marker_dict['marker_type'] = self.__class__.__name__
+        marker_dict['plot_on_signal'] = self._plot_on_signal
+
+        data_dict = {}
+        data_dict['x1'] = self.data['x1'].item().tolist()
+        data_dict['x2'] = self.data['x2'].item().tolist()
+        data_dict['y1'] = self.data['y1'].item().tolist()
+        data_dict['y2'] = self.data['y2'].item().tolist()
+        data_dict['text'] = self.data['text'].item().tolist()
+        data_dict['size'] = self.data['size'].item().tolist()
+        marker_dict['data'] = data_dict
+        return marker_dict
+
+    def _get_data_shape(self):
+        if self.data['x1'].flatten()[0].flatten()[0] is not None:
+            data_shape = np.array(self.data['x1'].item()).shape
+        elif self.data['x2'].flatten()[0].flatten()[0] is not None:
+            data_shape = np.array(self.data['x2'].item()).shape
+        elif self.data['y1'].flatten()[0].flatten()[0] is not None:
+            data_shape = np.array(self.data['y1'].item()).shape
+        elif self.data['y2'].flatten()[0].flatten()[0] is not None:
+            data_shape = np.array(self.data['y2'].item()).shape
```

Comment:
I believe `next(self.data[key][()].flat)` is faster than `self.data[key].flatten()[0].flatten()[0]` and gives the same output (but please check that it is the case).
I would also iterate over the keywords, not do the lookup twice.

Overall this should be equivalent:
```python
data_shape = None
for key in ('x1', 'x2', 'y1', 'y2):
    ar = self.data[key][()]
    if next(ar.flat) is not None:
        data_shape = ar.shape
        break
if data_shape is None:
    raise ValueError("None of the coordinates have value")
else:
    return data_shape
```